types/mparticle__web-sdk/index.d.ts

```diff
@@ -0,0 +1,811 @@
+    const logCheckout: LogCheckout;
+    const logImpression: LogImpression;
+    const logProductAction: LogProductAction;
+    const logPromotion: LogPromotion;
+    const logPurchase: LogPurchase;
+    /**
+     *
+     * @deprecated logRefund has been deprecated
+     */
+    const logRefund: LogRefund;
+    const setCurrencyCode: SetCurrencyCode;
+    // expandCommerceEvent function for internal use
+
+    const Cart: Cart;
+}
+
+export interface IdentifyRequest {
+    userIdentities: UserIdentities;
+}
+
+export type MPID = string;
+export interface User {
+    getUserIdentities: () => UserIdentities;
+    getMPID: () => MPID;
+    setUserTag: (tag: string) => void;
+    removeUserTag: (tag: string) => void;
+    setUserAttribute: (key: string, value: string) => void;
+    setUserAttributes: (attributeObject: Record<string, unknown>) => void;
+    removeUserAttribute: (key: string) => void;
+    setUserAttributeList: (key: string, value: UserAttributesValue[]) => void;
+    removeAllUserAttributes: () => void;
+    getUserAttributesLists: () => Record<string, UserAttributesValue[]>;
+    getAllUserAttributes: () => AllUserAttributes;
+    /**
+     *
+     * @deprecated Cart persistence in mParticle has been deprecated
+     */
+    getCart: () => Cart;
+    getConsentState: () => ConsentState;
+    setConsentState: (ConsentState: ConsentState) => void;
+    isLoggedIn: () => boolean;
+    getLastSeenTime: () => number;
+    getFirstSeenTime: () => number;
+}
+export type UserAttributesValue = string | number | boolean | null;
+export type AllUserAttributes = Record<
+    string,
+    UserAttributesValue | UserAttributesValue[]
+>;
+export interface UserIdentities {
+    customerid?: string;
+    email?: string;
+    other?: string;
+    other2?: string;
+    other3?: string;
+    other4?: string;
+    other5?: string;
+    other6?: string;
+    other7?: string;
+    other8?: string;
+    other9?: string;
+    other10?: string;
+    mobile_number?: string;
+    phone_number_2?: string;
+    phone_number_3?: string;
+    facebook?: string;
+    facebookcustomaudienceid?: string;
+    google?: string;
+    twitter?: string;
+    microsoft?: string;
+    yahoo?: string;
+}
+
+interface Cart {
+    /**
+     *
+     * @deprecated Cart persistence in mParticle has been deprecated. Please use mParticle.eCommerce.logProductAction(mParticle.ProductActionType.AddToCart, [products])
+     */
+    add: (product: Product, logEventBoolean?: boolean) => void;
+    /**
+     *
+     * @deprecated Cart persistence in mParticle has been deprecated. Please use mParticle.eCommerce.logProductAction(mParticle.ProductActionType.RemoveFromCart, [products])
+     */
+    remove: (product: Product, logEventBoolean?: boolean) => void;
+    /**
+     *
+     * @deprecated Cart persistence in mParticle has been deprecated.
+     */
+    clear: () => void;
+}
+
+export interface Product {
+    name: string;
```

Comment:
Hi @rmi2218 !, qq, shouldn't names here start with capital letter?, i.e: `Name: string`, etc 
https://docs.mparticle.com/developers/sdk/web/commerce-tracking/
<img width="427" alt="image" src="https://user-images.githubusercontent.com/433609/157109488-7c8ad38d-435c-41f9-915f-965629b1746a.png">

Reply:
@jaanauati  - this has been resolved in this pr - https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59832 and is now available as [version 2.15.1](https://www.npmjs.com/package/@types/mparticle__web-sdk) for @types/mparticle__web-sdk